clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -50,7 +50,7 @@
 
     // This matches URIs of formats: host:port and protocol:\\host:port
     // IPv6 is supported with [ip] pattern
-    private static final Pattern HOST_PORT_PATTERN = Pattern.compile(".*?\\[?([0-9a-zA-Z\\-.:]*)\\]?:([0-9]+)");
+    private static final Pattern HOST_PORT_PATTERN = Pattern.compile(".*?(?:(?:\\[([0-9a-zA-Z:]*)\\])|([0-9a-zA-Z\\-.]*))(?::([0-9]+))?");
```

Comment:
The issue with having only one type was inputs like: "127.0.0.1:8000" resulted in host: "127.0.0.1:8000" port: null matches as the we allowed ':' characters in the host part. Now we only allow the ':' character if it is surrounded with [] brackets.

I will check the remaining regex statements to see what can I do there.